sqllogictest-bin/src/main.rs

```diff
@@ -112,6 +113,18 @@ struct Opt {
     /// The engine name is a label by default.
     #[clap(long = "label")]
     labels: Vec<String>,
+
+    /// Partition ID for sharding the test files. When used with `partition_count`,
+    /// divides the test files into shards based on the hash of the file path.
+    ///
+    /// Useful for running tests in parallel across multiple CI jobs. Currently
+    /// automatically configured in Buildkite.
+    #[clap(long, env = "BUILDKITE_PARALLEL_JOB")]
```

Comment:
Seems we can only set on `env` name, but not many `env_aliases`. There's an issue but not resolved yet: https://github.com/clap-rs/clap/issues/5447.

I think we can have a adaptor layer to convert such CI platform env vars to our standard form like `SQLLOGICTEST_PARALLEL_JOB`.

Reply:
Added a new function called before `Opt::get_matches` to adapt env vars.